### PR TITLE
Changes

### PR DESCRIPTION
**Pull request by alekhyam94**
1) Including length in tabular form.
2) Sorting by score when evalues are same.

### CODE REVIEW

File: lib/sequenceserver/blast/query.rb [hunk 1]
```diff
@@ -17,7 +17,8 @@ def initialize(*args)
       end
 
       def sort_hits_by_evalue!
-        @hits = hits.sort_by(&:evalue)
+        # change made here
+        @hits = hits.sort_by{|h| [h.evalue,h.score]}
```
Comment (vivekiitkgp) [inline lib/sequenceserver/blast/query.rb]: Space after `,`.

**Comment by vivekiitkgp** [comment]
@yeban How important do we find the `hit.length` value? I'd like to discuss it a bit before merging.

**Comment by yeban** [comment]
@alekhyam94 I would prefer the length column be placed after score and evalue. Also, need to put a space after `,` as @vivekiitkgp pointed out.

@vivekiitkgp I would have benefitted from this just last week when I was analysing the results of annotation liftover. And as I have mentioned in #145 it's helpful to quickly get an overview of length of the hits for manual curation.

**Comment by vivekiitkgp** [comment]
While there is indeed no harm in doing this, these numbers are often quite large for a decent database and might occupy too much space (not a value for space trade).

Also, when we consider the immediate use case of Sequenceserver, if I'm blasting a query against a number of databases, the most important thing I'm concerned is with the quality of hits (it doesn't really matter from how big database do they come from, at least, during the start). I can always navigate to the respective hit and examine the database size if needed.

This is what I had in mind.

**Comment by yeban** [comment]
Because you say "big database" I'm wondering if you are misunderstanding the issue. We are talking hit length here, not database size. Again, what's important is what the hits are saying about my query sequence. I think when blasting proteins and transcripts, length of the hit is almost always an important metric. For instance, if BLAST a predicted protein sequence against Uniprot and I see that all the strong hits are longer than my query sequence, it could mean the predicted protein is incorrect. And that is a significant use case of BLAST & SequenceServer.

File: lib/sequenceserver/blast/query.rb [hunk 2]
```diff
@@ -17,7 +17,8 @@ def initialize(*args)
       end
 
       def sort_hits_by_evalue!
-        @hits = hits.sort_by(&:evalue)
+        # change made here
+        @hits = hits.sort_by{|h| [h.evalue, h.score]}
```
Comment (yeban) [inline lib/sequenceserver/blast/query.rb]: Very nicely done.

**Comment by yeban** [comment]
Thanks @alekhyam94. Merged. You forgot to change table headers when you moved around the length column. I took care of that (ac6dea5302c8a7f622d74c1a53b9f9c5df10d453).

**Comment by vivekiitkgp** [comment]
> Because you say "big database" I'm wondering if you are misunderstanding the issue.

Umm... No. I'm not. Of course `big database` was a typo but just blast that `multi_hsp` query against the `Si_gnF` database and you'll see that a lot of hits have `hit.length` in the range of 2 * 10 ^ 6. Clearly, that adds too little information.

> if BLAST a predicted protein sequence against Uniprot and I see that all the strong hits are longer than my query sequence, it could mean the predicted protein is incorrect.

I don't think that's a correct inference. In my idea, that doesn't mean anything significant at all. For example, in the previous quote the hit length is in million range but still there might be an exact match to my query.

Correct me if I'm wrong.

**Comment by yeban** [comment]
Protein sequences are much smaller. And my example considers protein-protein BLAST(p).

**Comment by vivekiitkgp** [comment]
Alright. Does no harm if you insist.

**Pull request merged**